### ISSUE TITLE
`commander` -V option does not work on Windows

### ISSUE THREAD

**Issue by vu-bui** [jxcore/jxcore#764]
I have the following code snippet:
```javascript
#!/usr/bin/env node

'use strict';

var program = require('commander');
var version = require('./package.json').version;

console.log(version);
console.log(process.argv);

program
  .version(version)
  .parse(process.argv);
```
Here is the output when running in `node` compared to `jx`:
```
{ test }  » node index.js -V
1.0.0
[ 'C:\\Users\\vubui\\Downloads\\node\\node.exe',
  'D:\\development\\test\\index.js',
  '-V' ]
1.0.0
```
```
{ test }  » jx index.js -V
1.0.0
```
The above code does work on OSX and Linux. All of the `jx` installations yield the same version number:
```
{ test }  » jx -jxv
v0.3.1.0
{ test }  » jx -jsv
Google V8 v3.14.5.9
```

**Comment by vu-bui** [comment]
I found that it's because I'm using `cygwin`. Changing to Command Prompt or PowerShell should resolve this. I don't know if this is really on issue or not.

**Comment by ktrzeciaknubisa** [comment]
Well, if for node that works, it might be an issue of JXcore. I wonder it this is just that `-V` does not work or `console.log(process.argv)` is not getting flushed to the console?

**Comment by ktrzeciaknubisa** [comment]
Yes, it is how I expected. If you execute your `index.js` through `child_process` it behaves the same on Windows:

**start.js** :
```js
var cp = require('child_process');

cp.exec('"' + process.execPath + '" ./index.js -V', function (error, stdout, stderr) {
  console.log(error || stdout + '');
});
```

But this is an old problem https://github.com/jxcore/jxcore/issues/543 and also was present in older versions of node. Which node are you using right now?

**Comment by vu-bui** [comment]
Sorry for late reply, this only happens with the `-V` option. Without that, the `console.log(process.argv)` prints output as normal. And I'm using the latest stable version of node, which is 5.3.0.

**Comment by ktrzeciaknubisa** [comment]
> this only happens with the -V option

Yes, because when `process.exit()` is triggered by `-V` ([index.js#L825](https://github.com/tj/commander.js/blob/master/index.js#L825)) then the problem occurs. You can achieve the same without any `commander` dependency, e.g. :

```js
console.log('1');
console.log('2');
process.exit();
```

**Comment by vu-bui** [comment]
I get it now. So it's because the process does not properly flush the output before exiting, and the reason why I didn't get that problem on node is because I'm using the newer node version, which fixed the issue already. Thanks for your support!

**Comment by ktrzeciaknubisa** [comment]
Yes, exactly. You're welcome! :)

**Comment by obastemur** [comment]
This should be fixed by 0.3.1.1 closing this one as it is a duplicate now.
